test/com/google/enterprise/adaptor/ConfigTest.java

```diff
@@ -330,4 +332,107 @@ public void testPropertiesParseEscapeASlash6() throws Exception {
     config.load(configFile);
     assertEquals("\\\t\\\f", config.getValue("slash"));
   }
+
+  @Test
+  public void testValidValuesForTimeouts() throws Exception {
+    configFile.setFileContents("gsa.hostname=not_used\n"
+                               + "adaptor.incrementalPollPeriodSecs=1700\n"
+                               + "adaptor.docHeaderTimeoutSecs=50\n"
+                               + "adaptor.docContentTimeoutSecs=190\n"
+                               + "adaptor.heartbeatTimeoutSecs=\n");
```

Comment:
In this test, explicitly set heartbeat, then add another test that verifies the fallback value.